drivers/i2c/Kconfig.stm32

```diff
@@ -47,4 +47,12 @@ config I2C_STM32_COMBINED_INTERRUPT
 	depends on I2C_STM32_INTERRUPT
 	default y if SOC_SERIES_STM32F0X || SOC_SERIES_STM32L0X
 
+config I2C_STM32_TIMEOUT_USEC
+	int "STM32 MCU I2C transaction timeout, in microseconds"
+	depends on I2C_STM32_V1
```

Comment:
Should we change it to `depends on I2C_STM32_V1 && !I2C_STM32_INTERRUPT` as timeout option is available only in polling mode?